web-app/js/portal/filter/BooleanFilter.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.filter');
+
+Portal.filter.BooleanFilter = Ext.extend(Portal.filter.Filter, {
+
+    getSupportedGeoserverTypes: function() {
+
+        return ['boolean'];
```

Comment:
Good question! As I see it, there are basically two options here.

Option (1): We could have a mapping so that there is one place to go to find out which GeoServer types map to which filter types. That is essentially what we had [here](https://github.com/aodn/aodn-portal/blob/d4044e3de30ee35770e4dbd8b88c2e8ff9ef1b91/web-app/js/portal/filter/BaseFilterPanel.js#L131-L149). One method that knows all the GeoServer types that each Filter type can handle.

Option (2) is what I've tried here. Essentially we have the ability to "ask" each Filter type what GeoServer types it supports. I liked this approach because it keeps this information within the class it is applicable for. Filter.js still needs to have a list of it's subclasses (see https://github.com/aodn/aodn-portal/blob/1455eb6141e42c60c90191c563b5968d4f4c17c6/web-app/js/portal/filter/Filter.js#L90-L116) but at least it doesn't need to also know which types each subtype supports.

I'll be interested to see how this holds-up as the Portal evolves.